antarest/study/storage/variantstudy/model/command/update_binding_constraint.py

```diff
@@ -16,7 +25,189 @@
 from antarest.study.storage.variantstudy.model.command.icommand import MATCH_SIGNATURE_SEPARATOR, ICommand
 from antarest.study.storage.variantstudy.model.model import CommandDTO
 
-MatrixType = List[List[MatrixData]]
+MatrixType = t.List[t.List[MatrixData]]
+
+ALIAS_OPERATOR_MAP = {
+    BindingConstraintOperator.EQUAL: "eq",
+    BindingConstraintOperator.LESS: "lt",
+    BindingConstraintOperator.GREATER: "gt",
+}
+
+
+class MatrixInputTargetPaths(BaseModel, frozen=True, extra="forbid"):
+    """
+    Model used to store the input and target paths for matrices.
+    """
+
+    matrix_input_paths: t.List[Path] = Field(..., min_items=1, max_items=2)
+    matrix_target_paths: t.List[Path] = Field(..., min_items=1, max_items=2)
+
+
+def _infer_input_and_target_paths(
+    parent_node: FolderNode,
+    binding_constraint_id: str,
+    existing_operator: BindingConstraintOperator,
+    new_operator: BindingConstraintOperator,
+) -> MatrixInputTargetPaths:
+    """
+    Infer the target paths of the matrices to update according to the existing and new operators.
+
+    Args:
+        parent_node: the parent folder node
+        binding_constraint_id: the binding constraint ID
+        existing_operator: the existing operator
+        new_operator: the new operator
+
+    Returns:
+        the matrix input and target paths
+    """
+    # new and existing operators should be different
+    assert existing_operator != new_operator, "Existing and new operators should be different"
```

Comment:
I think you'll change this when handling operator equality

Reply:
It might be stupid but as you have many ifs in functions, this increase the complexity. So I had to handle the equality out of the function. The downside, you should enforce such case, just because you never know what a developer could do.